package/MDAnalysis/core/selection.py

```diff
@@ -838,9 +822,11 @@ def apply(self, group):
         return group[mask].unique
 
 
-class ResnumSelection(RangeSelection):
-    token = 'resnum'
-    field = 'resnums'
+class FloatRangeSelection(RangeSelection):
+    """Range selection for float values"""
+
+    pattern = r"(-?\d*\.?\d*)\s*(?:[:-]|to)\s*(-?\d*\.?\d*)"
```

Comment:
Document that engineering notation is not supported.

Reply:
We do now. Also moved to one place for future enhancement of ranges and such.

```
FLOAT_PATTERN = "-?\d*\.?\d*(?:e[-+]?\d+)?"
INT_PATTERN = "-?\d+"
RANGE_PATTERN = "\s*(?:[:-]| to )\s*"
```